src/main/java/org/jenkins/tools/test/model/PluginRemoting.java

```diff
@@ -73,71 +72,34 @@ public PluginRemoting(String hpiRemoteUrl){
     public PluginRemoting(File pomFile){
         this.pomFile = pomFile;
     }
-	
-    private String retrievePomContent() throws PluginSourcesUnavailableException{
-        if (hpiRemoteUrl != null) {
-            return retrievePomContentFromHpi();
-        } else {
-            return retrievePomContentFromXmlFile();
-        }
-    }
 
-    private String retrievePomContentFromHpi() throws PluginSourcesUnavailableException {
+    private String retrievePomContent() {
         try {
-	    return retrievePomContentFromHpiRemoteUrl(new URL(hpiRemoteUrl));
-	} catch (MalformedURLException e) {
-	    LOGGER.log(Level.WARNING, "HPI reference {0} is not a remote URL", hpiRemoteUrl);
-	    return retrievePomContentFromHpiFileReference();
-	}
-    }
-
-    private String retrievePomContentFromHpiRemoteUrl(URL url) throws PluginSourcesUnavailableException {
-    	try {
-    	    return retrievePomContentFromInputStream(url.openStream());
-	} catch (IOException e) {
-	    LOGGER.log(Level.WARNING, "Failed to retrieve POM content from HPI remote URL", e);
-          throw new PluginSourcesUnavailableException("Failed to retrieve POM content from HPI remote URL", e);
-	}
-    }
-
-    private String retrievePomContentFromHpiFileReference() throws PluginSourcesUnavailableException {
-    	try {
-	    String fileReference = hpiRemoteUrl.replaceAll("jar:", "").replaceAll("!/name.hpi", "");
```

Comment:
Whatever this was trying to do, the implementation makes no sense ("name.hpi"??) and I did not hit this code path with a breakpoint, so it is dead code.